src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -656,6 +656,23 @@ public function create_item( $request ) {
 			}
 		}
 
+		/**
+		 * Filters the type of notification sent upon new user registration.
+		 *
+		 * This filter allows customization of the notification type when a new user is created via REST API.
+		 *
+		 * @param string $notify  Determines who gets notified. Accepts:
+		 *                        - 'admin' (default) or an empty string: Notify only the site administrator.
+		 *                        - 'user': Notify only the new user.
+		 *                        - 'both': Notify both admin and user.
+		 *                        - false: Disable notifications entirely.
+		 * @param int    $user_id User ID.
+		 */
+		$notify = apply_filters( 'rest_wp_user_created_notification', 'admin', $user_id );
+		if ( false !== $notify ) {
```

Comment:
I think this is slightly better, as it avoids the possibility of sending invalid values in:

```suggestion
		if ( in_array( $notify, array( 'admin', 'user', 'both' ), true ) ) {
```